### PR TITLE
Add some accessors and update XTL (small)

### PR DESCRIPTION
**Pull request by gridley**
Hey all,

So, last time I updated XTL, I thought a change I was trying to make there had gone through that lets it work with the latest CUDA compiler, but it seems it had in fact not yet gone through. Now, it has. That change is xtensor-stack/xtl#188 and mpark/variant#73. So, firstly, this PR takes XTL up to its 0.6.13 release.

Second, I added some methods for obtaining observing pointers to some unique_pointers in order to adapt data from CPU to GPU. I know that raw pointers are generally desirable to avoid, but the core guidelines say that they're OK in the context of being observing pointers, which they are here. So, yeah. Please let me add this :smile: 

https://stackoverflow.com/questions/38901846/pointing-to-the-content-of-stdunique-ptr

### CODE REVIEW

File: include/openmc/distribution_spatial.h [hunk 1]
```diff
@@ -112,6 +129,8 @@ class SpatialPoint : public SpatialDistribution {
   //! \param seed Pseudorandom number seed pointer
   //! \return Sampled position
   Position sample(uint64_t* seed) const;
+
+  auto r() const { return r_; }
```
Comment (paulromano) [inline include/openmc/distribution_spatial.h]:
```suggestion
  Position r() const { return r_; }
```

File: include/openmc/distribution_multi.h [hunk 2]
```diff
@@ -43,6 +43,11 @@ class PolarAzimuthal : public UnitSphereDistribution {
   //! \param seed Pseudorandom number seed pointer
   //! \return Direction sampled
   Direction sample(uint64_t* seed) const;
+
+  // Observing pointers
+  auto mu() const { return mu_.get(); }
+  auto phi() const { return phi_.get(); }
```
Comment (paulromano) [inline include/openmc/distribution_multi.h]: I personally like to see the return type here
```suggestion
  Distribution* mu() const { return mu_.get(); }
  Distribution* phi() const { return phi_.get(); }
```

File: include/openmc/distribution_spatial.h [hunk 3]
```diff
@@ -32,6 +32,11 @@ class CartesianIndependent : public SpatialDistribution {
   //! \param seed Pseudorandom number seed pointer
   //! \return Sampled position
   Position sample(uint64_t* seed) const;
+
+  // Observer pointers
+  auto x() const { return x_.get(); }
+  auto y() const { return x_.get(); }
+  auto z() const { return x_.get(); }
```
Comment (paulromano) [inline include/openmc/distribution_spatial.h]:
```suggestion
  Distribution* x() const { return x_.get(); }
  Distribution* y() const { return x_.get(); }
  Distribution* z() const { return x_.get(); }
```

File: include/openmc/distribution_spatial.h [hunk 4]
```diff
@@ -50,6 +55,11 @@ class CylindricalIndependent : public SpatialDistribution {
   //! \param seed Pseudorandom number seed pointer
   //! \return Sampled position
   Position sample(uint64_t* seed) const;
+  
+  auto r() const { return r_.get(); }
+  auto phi() const { return phi_.get(); }
+  auto z() const { return z_.get(); }
+  auto origin() const { return origin_; }
```
Comment (paulromano) [inline include/openmc/distribution_spatial.h]:
```suggestion
  Distribution* r() const { return r_.get(); }
  Distribution* phi() const { return phi_.get(); }
  Distribution* z() const { return z_.get(); }
  Position origin() const { return origin_; }
```

File: include/openmc/distribution_spatial.h [hunk 5]
```diff
@@ -70,6 +80,11 @@ class SphericalIndependent : public SpatialDistribution {
   //! \param seed Pseudorandom number seed pointer
   //! \return Sampled position
   Position sample(uint64_t* seed) const;
+
+  auto r() const { return r_.get(); }
+  auto theta() const { return theta_.get(); }
+  auto phi() const { return phi_.get(); }
+  auto origin () const { return origin_; }
```
Comment (paulromano) [inline include/openmc/distribution_spatial.h]:
```suggestion
  Distribution* r() const { return r_.get(); }
  Distribution* theta() const { return theta_.get(); }
  Distribution* phi() const { return phi_.get(); }
  Position origin () const { return origin_; }
```

File: include/openmc/source.h [hunk 6]
```diff
@@ -43,7 +43,14 @@ class SourceDistribution {
   Particle::Bank sample(uint64_t* seed) const;
 
   // Properties
+  Particle::Type particle_type() const { return particle_; }
   double strength() const { return strength_; }
+
+  // Make observing pointers available
+  auto space() const { return space_.get(); }
+  auto angle() const { return angle_.get(); }
+  auto energy() const { return energy_.get(); }
```
Comment (paulromano) [inline include/openmc/source.h]:
```suggestion
  SpatialDistribution* space() const { return space_.get(); }
  UnitSphereDistribution* angle() const { return angle_.get(); }
  Distribution* energy() const { return energy_.get(); }
```

**Comment by gridley** [comment]
Thanks for those suggested changes Paul. Thinking about it, this would make it easier on future developers who just want to look at the header to know the interface to stuff. Also, sheesh, the "commit changes" feature is awesome. So convenient! Thanks for your prompt response!

**Comment by gridley** [comment]
This should be good-to-go, whenever is good for you @paulromano.

**Pull request merged**